bentoml/yatai/__init__.py

```diff
@@ -75,16 +75,35 @@ def ApplyDeployment(self, request, context=None):
                 request.deployment.namespace or self.default_namespace
             )
 
-            # TODO: validate deployment config
+            deployment_orm = self.deployment_store.get(
+                request.deployment.name, request.deployment.namespace
+            )
+            if deployment_orm:
+                # check deployment platform
+                if deployment_orm.spec.operator != request.deployment.spec.operator:
+                    return ApplyDeploymentResponse(
+                        status=Status.ABORTED(
+                            'New deployment spec has different platform from existing one.'
+                        )
+                    )
+                previous_deployment = deployment_orm
 
-            # create or update deployment spec record
-            self.deployment_store.insert_or_update(request.deployment)
+                with self.deployment_store.update_deployment(
```

Comment:
prefer to use insert_or_update - annotation & labels can be updated in this call as well

Reply:
I will do more checks here and add information if it is available.  I end up with using update because I don’t want override state

Reply:
just do `request.deployment.state = ...` and `insert_or_update` at the end